### ISSUE TITLE
MQTT Dev Service fails to start with podman 4.6.2 and crun 1.9 installed

### ISSUE THREAD

**Issue by ttaylor249** [quarkusio/quarkus#35860]
### Describe the bug

I have an application I've been working on for the past month with no issues.  Recently, my system's container runtime (crun) was upgraded from version 1.8.7 to 1.9.

Since updating, any attempts to run my app in dev or test mode result in failures trying to start the MQTT container instance.

NOTE: I suspect this failure would happen with any Dev Services launched container, but I've only been testing with the smallrye-reactive-messaging-mqtt extension and it's dev services at the moment

Running `quarkus dev` produces this error:
```
2023-09-11 15:22:44,685 DEBUG [org.tes.uti.MountableFile] (build-10) Copying resource mosquitto.conf from JAR file /home/ttaylor/.m2/repository/io/quarkus/quarkus-smallrye-reactive-messaging-mqtt-deployment/3.3.2/quarkus-smallrye-reactive-messaging-mqtt-deployment-3.3.2.jar
2023-09-11 15:22:44,710 DEBUG [org.tes.sha.com.git.doc.cor.com.AbstrDockerCmd] (build-10) Cmd: org.testcontainers.shaded.com.github.dockerjava.core.command.CopyArchiveToContainerCmdImpl@31db78a5[cp ,-a=false ,<null>, ,861aca0e6e3677e2d592c3a1e094f941456e1289473fb388d7aea62223524c10,:,/]
2023-09-11 15:22:44,756 INFO  [tc.ecl.0.15] (build-10) Container eclipse-mosquitto:2.0.15 is starting: 861aca0e6e3677e2d592c3a1e094f941456e1289473fb388d7aea62223524c10
2023-09-11 15:22:44,757 DEBUG [org.tes.sha.com.git.doc.cor.com.AbstrDockerCmd] (build-10) Cmd: 861aca0e6e3677e2d592c3a1e094f941456e1289473fb388d7aea62223524c10
2023-09-11 15:22:45,084 ERROR [tc.ecl.0.15] (build-10) Could not start container: com.github.dockerjava.api.exception.InternalServerErrorException: Status 500: {"cause":"OCI permission denied","message":"crun: write to `/proc/self/oom_score_adj`: Permission denied: OCI permission denied","response":500}

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:247)
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:102)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.StartContainerCmdExec.execute(StartContainerCmdExec.java:31)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.StartContainerCmdExec.execute(StartContainerCmdExec.java:13)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.StartContainerCmdImpl.exec(StartContainerCmdImpl.java:43)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:441)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:344)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.lambda$startMqttBroker$1(MqttDevServicesProcessor.java:167)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttBroker(MqttDevServicesProcessor.java:185)
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttDevService(MqttDevServicesProcessor.java:81)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:858)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

The system log contains the following errors when this happens:
```
Sep 11 15:22:44 mm248056-pc conmon[87995]: conmon 861aca0e6e3677e2d592 <nwarn>: runtime stderr: write to `/proc/self/oom_score_adj`: Permission denied
Sep 11 15:22:45 mm248056-pc podman[87230]: time="2023-09-11T15:22:45-04:00" level=info msg="Request Failed(Internal Server Error): crun: write to `/proc/self/oom_score_adj`: Permission denied: OCI permission denied"
```

The container that is trying to be run is:  docker.io/library/eclipse-mosquitto:2.0.15

Running this container manually via: `podman run -it --rm docker.io/library/eclipse-mosquitto:2.0.15` works fine but fails when started by quarkus devservices.

Downgrading the crun package to version 1.8.7 restored the previous behavior.

The NEWS file [1] for crun 1.9 has this entry:
```
- linux: fix check for oom_score_adj.  Write the oom_score_adj file
  even when the new value is 0.
```

[1] https://github.com/containers/crun/blob/1.9/NEWS

### Expected behavior

The eclipse-mosquitto MQTT Dev Services container instance should start successfully and my app should run.

There should be a running mosquitto MQTT container instance similar to this:
```
CONTAINER ID  IMAGE                                       COMMAND               CREATED         STATUS                PORTS                    NAMES
bdca1ae65699  docker.io/library/eclipse-mosquitto:2.0.15  /usr/sbin/mosquit...  18 seconds ago  Up 18 seconds         0.0.0.0:39219->1883/tcp  loving_yalow
```

### Actual behavior

Quarkus Dev Services fails to start the eclipse-mosquitto MQTT container instance with the error:
```
2023-09-11 15:22:45,084 ERROR [tc.ecl.0.15] (build-10) Could not start container: com.github.dockerjava.api.exception.InternalServerErrorException: Status 500: {"cause":"OCI permission denied","message":"crun: write to `/proc/self/oom_score_adj`: Permission denied: OCI permission denied","response":500}

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:247)
```

### How to Reproduce?

Steps to reproduce:
1) Create a new quarkus app: `quarkus create app --extension=smallrye-reactive-messaging-mqtt mqtt-test`
2) Create a file named `MqttTest.java` at `src/main/java/example` with the following contents:
```
package example;

import org.eclipse.microprofile.reactive.messaging.Incoming;
import org.jboss.logging.Logger;

import io.vertx.core.json.JsonObject;
import jakarta.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class MqttTest {
    private static final Logger LOG = Logger.getLogger(MqttTest.class);
    
    @Incoming("updates")
    public void updates(byte[] rawMsg) {
        JsonObject json = new JsonObject(new String(rawMsg));
        LOG.debug("Got update message: " + json.toString());
    }

}
```
3) Run `quarkus dev`
4) In another terminal window, run `podman ps`.  There should be a running instance of the mosquitto container, like this:
```
CONTAINER ID  IMAGE                                       COMMAND               CREATED         STATUS                PORTS                    NAMES
bdca1ae65699  docker.io/library/eclipse-mosquitto:2.0.15  /usr/sbin/mosquit...  18 seconds ago  Up 18 seconds         0.0.0.0:39219->1883/tcp  loving_yalow
```
but if crun 1.9 is installed (and is the engine used by podman) then a permission exception is thrown.

### Output of `uname -a` or `ver`

Linux mm248056-pc 6.4.14-200.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Sep  2 16:36:06 UTC 2023 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.8" 2023-07-18 OpenJDK Runtime Environment (Red_Hat-17.0.8.0.7-1.fc38) (build 17.0.8+7) OpenJDK 64-Bit Server VM (Red_Hat-17.0.8.0.7-1.fc38) (build 17.0.8+7, mixed mode, sharing)

### Quarkus version or git rev

3.3.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.3 (21122926829f1ead511c958d89bd2f672198ae9f) Maven home: /home/ttaylor/.m2/wrapper/dists/apache-maven-3.9.3-bin/326f10f4/apache-maven-3.9.3 Java version: 17.0.8, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-17-openjdk-17.0.8.0.7-1.fc38.x86_64 Default locale: en_US, platform encoding: UTF-8 OS name: "linux", version: "6.4.14-200.fc38.x86_64", arch: "amd64", family: "unix"

### Additional information

I'm running Fedora 38.
Podman version 4.6.2 in rootless mode.
crun version 1.8.7-1.fc38 works
crun version 1.9-1.fc38 does not

**Comment by snazy** [comment]
I'm experiencing the same issue - but completely unrelated to Quarkus

**Comment by snazy** [comment]
I noticed, that I've updated `crun` on my machine yesterday. You maybe as well?

**Comment by ozangunalp** [comment]
As mentioned this is unrelated to Quarkus and there is an open issue in crun : https://github.com/containers/crun/issues/1302 
I suggest you to contribute your findings to the original issue

**Comment by snazy** [comment]
Opened https://github.com/docker-java/docker-java/issues/2201

**Comment by snazy** [comment]
Hm - I suspect, the JSON parsing of the Docker request is odd. The change that causes the error checks whether that `OomScoreAdj` is present - and docker-java sends `null`. But even if I tweak docker-java to not send that value at all, the error still happens.

**Comment by xstefank** [comment]
just fyi, on Fedora 38, I fixed it with `sudo dnf install crun-1.8.3-2.fc38.x86_64`.
